plugins/arElasticSearchPlugin/lib/model/arElasticSearchInformationObject.class.php

```diff
@@ -82,7 +103,9 @@ public function recursivelyAddInformationObjects($parentId, $totalRows, $options
   public static function update($object, $options = array())
   {
     // Update description
-    $node = new arElasticSearchInformationObjectPdo($object->id);
+    $node = new arElasticSearchInformationObjectPdo($object->id,
+      array('terms' => self::loadTerms(array('taxonomyIds' => array(QubitTaxonomy::SUBJECT_ID, QubitTaxonomy::PLACE_ID, QubitTaxonomy::GENRE_ID)))));
```

Comment:
How much can this take when there is a lot of those terms? This is executed sometimes synchronously in a request cycle and it may be too much for a simple description.

I'm not completely sure, but I think `$object` is always a `QubitInformationObject` in here. If that's the case, we could at least check if it has any object - term relations before loading the terms lookup.

Reply:
I have added logic to not load the terms list if there are no otr records for this object. This will *not* apply if descendants are set to be indexed too - in this case I will load the terms list in case there are terms attached to the descendants.

Reply:
I have added a check here to stop this if there are no otr records for a given object. I have made the object's term list property static, so that if it does encounter a descendant that requires this list, it will be loaded at that time and preserved for later calls.  It should only be loaded once.  

When loading ~360K terms, using the update() method and loading the terms seems to take the same amount of time as the old logic.

Reply:
Awesome, thanks!!